app/controllers/password_resets_controller.rb

```diff
@@ -0,0 +1,23 @@
+class PasswordResetsController < ApplicationController
+  def show
+    user = User.where(token: params[:id]).first
+    if user
+      @token = user.token
+    else
+      redirect_to expired_token_path
+    end
+  end
+
+  def create
+    user = User.where(token: params[:token]).first
```

Comment:
Again, we want to use `find_by` rather than `where().first` - make sure you know the difference.